slither/visitors/slithir/expression_to_slithir.py

```diff
@@ -211,7 +212,31 @@ def _post_assignement_operation(self, expression):
                 # a = b = 1;
                 set_val(expression, left)
 
+    def _attempt_constant_folding(self, expression):
+        try:
+            const_fold = ConstantFolding(expression, expression.type)
+        except (NotConstant, AttributeError):
```

Comment:
Catching an attribute error here seems like it could hide a bug from us.

If somewhere in the `ConstantFolding` implementation an attribute name is spelled incorrectly, the error will get caught here and we will fail to fold. That doesn't have anything to do with the expression being unfoldable; instead, it's a bug that we should fix. Allowing the stack trace at the point the AttributeError was thrown to reach the top level will encourage us to fix the bug.

Reply:
Practically, `ConstantFolding` raises `AttributeError` when it encounters `bytes` or a `string`. It is more like a `NotImplementedError` sort of thing in my opinion.
I agree that at some point this incompleteness in `ConstantFolding` should be addressed but I am not sure it is in the scope of this PR.

Reply:
We would have to convince the Slither developers to fix this. Since I'm not sure if they will want to change it, I'll just mark this comment as resolved.